File: LeetCode/Easy/Problems 951 - 1000/Problem 977/p977Sol3.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        length = len(nums)
        start = 0
        end = length - 1
        insertIndex = end
        sol = [None] * length #Create place holder list
        
        while start <= end:
            leftNum = abs(nums[start])
            rightNum = abs(nums[end])
            
            if leftNum >= rightNum:
                sol[insertIndex] = leftNum ** 2
                insertIndex -= 1
                start += 1
            else:
                sol[insertIndex] = rightNum ** 2
                insertIndex -= 1
                end -= 1
                
        return sol

File: LeetCode/Easy/Problems 951 - 1000/Problem 977/p977Sol3.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* sortedSquares(int* nums, int numsSize, int* returnSize){
    int start = 0;
    int end = numsSize - 1;
    int insertIndex = end;
    *returnSize = numsSize;
    int* sol = malloc(numsSize * sizeof(int));
    int leftNum, rightNum;
    
    
    while (start <= end) {
        //No need to include <stdlib.h> for abs() because leetcode already includes it
        leftNum = abs(nums[start]);
        rightNum = abs(nums[end]);
        
        if (leftNum >= rightNum) {
            sol[insertIndex] = leftNum * leftNum;
            insertIndex--;
            start++;
        }
        else {
            sol[insertIndex] = rightNum * rightNum;
            insertIndex--;
            end--;
        } 
    }
    
    return sol;
}